plugins/experimental/ssl_session_reuse/src/publish.cc

```diff
@@ -117,8 +117,8 @@ RedisPublisher::is_good()
 ::redisContext *
 RedisPublisher::setup_connection(const RedisEndpoint &re)
 {
-  ::pthread_t my_id = ::pthread_self();
-  TSDebug(PLUGIN, "RedisPublisher::setup_connection: Called by threadId: %lx", my_id);
+  auto my_id = (unsigned long long)::pthread_self();
```

Comment:
Why not C++ style cast?

Reply:
Linux wants to have static_cast (changing the size of the unsigned long) and FreeBSD/macOS wants to have a reinterpret_cast (change from a pointer to an unsigned long long).  So we have to use a C style cast.

Reply:
OK, I thought reinterpret_cast works for the both cases.

Reply:
Yeah, I thought it would have taken care of both cases too.  That is what I did in my first commit for the PR, but it failed for the Linux OSes.